File: JavaImpl/src/main/java/org/example/BigONotation/LinearSearchExample.java
package org.example.BigONotation;

public class LinearSearchExample {
    public static void main(String[] args) {
        int[] array = {1, 2, 3, 4, 5};
        boolean found = contains(array, 3);
        System.out.println("Found: " + found); // O(n)
    }

    public static boolean contains(int[] array, int target) {
        for (int num : array) {
            if (num == target) return true;
        }
        return false; // العنصر مش موجود
    }
}

File: KotlinImpl/src/main/kotlin/BigONotation/LinearSearchExample.kt
package org.example.BigONotation

fun main() {
    val array = intArrayOf(1, 2, 3, 4, 5)
    val found = contains(array, 3)
    println("Found: $found") // O(n)
}

fun contains(array: IntArray, target: Int): Boolean {
    for (num in array) {
        if (num == target) return true
    }
    return false // العنصر مش موجود
}
